azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CFilterSignUpOrInHandler.java

```diff
@@ -25,42 +31,94 @@
 public class AADB2CFilterSignUpOrInHandler extends AbstractAADB2CFilterScenarioHandler
         implements AADB2CFilterScenarioHandler {
 
+    private final RedirectStrategy redirectStrategy = new DefaultRedirectStrategy();
+
     /**
      * Mapping configuration URL to ${@link AADB2CJWTProcessor}.
      */
-    private final ConcurrentMap<String, AADB2CJWTProcessor> urlToJwtParser = new ConcurrentHashMap<>();
+    private static final ConcurrentMap<String, AADB2CJWTProcessor> URL_TO_JWT_PARSER = new ConcurrentHashMap<>();
 
     /**
      * Get cached instance of ${@link AADB2CJWTProcessor} from given url, or create new one.
      *
      * @param url        of the configuration.
      * @param properties of ${@link AADB2CProperties}.
      * @return the instance of ${@link AADB2CJWTProcessor}.
-     * @throws AADB2CAuthenticationException when failed to create ${@link AADB2CJWTProcessor} instance.
      */
-    private AADB2CJWTProcessor getAADB2CJwtParser(@URL String url, @NonNull AADB2CProperties properties)
-            throws AADB2CAuthenticationException {
-        urlToJwtParser.putIfAbsent(url, new AADB2CJWTProcessor(url, properties));
+    private AADB2CJWTProcessor getAADB2CJwtProcessor(@URL String url, @NonNull AADB2CProperties properties) {
+        return URL_TO_JWT_PARSER.computeIfAbsent(url, k -> new AADB2CJWTProcessor(k, properties));
+    }
 
-        return urlToJwtParser.get(url);
+    private void validateState(String state, HttpServletRequest request) throws AADB2CAuthenticationException {
+//        Assert.hasText(state, "state should contains text.");
+//
+//        final String requestURL = request.getSession().getAttribute(AADB2CURL.ATTRIBUTE_STATE).toString();
+//
+//        if (!state.equals(requestURL)) {
+//            throw new AADB2CAuthenticationException("The reply state has unexpected content: " + state);
+//        }
     }
 
-    @Override
-    public void handle(HttpServletRequest request, HttpServletResponse response, AADB2CProperties properties)
-            throws AADB2CAuthenticationException {
-        super.validateAuthentication(request);
+    private void validateNonce(String nonce, HttpServletRequest request) throws AADB2CAuthenticationException {
+//        Assert.hasText(nonce, "nonce should contains text.");
+//
+//        final String expectedNonce = request.getSession().getAttribute(AADB2CURL.ATTRIBUTE_NONCE).toString();
+//
+//        if (!nonce.equals(expectedNonce)) {
+//            throw new AADB2CAuthenticationException("The claim nonce has unexpected content: " + nonce);
+//        }
+    }
 
-        Assert.notNull(properties, "AADB2CProperties should not be null.");
+    private boolean isAuthenticated(Authentication auth) {
```

Comment:
could be static

Reply:
will move to abstract class, as other implementation may leverage this.